pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -125,6 +127,7 @@ func (nc *NamespaceController) startCaBundleWatcher(stop <-chan struct{}) {
 // reconcileCACert will reconcile the ca root cert configmap for the specified namespace
 // If the configmap is not found, it will be created.
 // If the namespace is filtered out by discovery selector, the configmap will be deleted.
+// TODO: find the code where delete happens or update the previous line.
```

Comment:
I also couldn't find where the delete happens after a quick search, but I think this scenario might be tested in `namespacecontroller_test.go` `TestNamespaceControllerDiscovery`.